app/workers/send_user_invitation_worker.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SendUserInvitationWorker < ApplicationJob
+  queue_as :default
+
+  RETRY_ERRORS = [
+    OpenSSL::SSL::SSLError,
+    Net::SMTPAuthenticationError,
+    Net::SMTPFatalError,
+    Net::SMTPServerBusy,
+    Net::SMTPSyntaxError,
+    Net::SMTPUnknownError,
+    Net::SMTPUnsupportedCommand,
+    SocketError,
+  ].freeze
+
+  DISCARD_ERRORS = [
+    ActiveRecord::RecordNotFound
+  ].freeze
+
+  def perform(invitation_id)
+    invitation = Invitation.find(invitation_id)
+
+    mailer = invitation.account.provider? ? ProviderInvitationMailer : InvitationMailer
+    mailer.invitation(invitation).deliver_now!
+
+    invitation.update!(sent_at: Time.zone.now)
```

Comment:
The `update!` _should_ never fail, but if it does (for example, if the DB is down for a while), then it will retry the whole job, and it will send the email again.

Reply:
I'll add `ActiveRecord::ActiveRecordError` to the _discard errors_.

Reply:
Then it is better just doing `invitation.update(sent_at: Time.zone.now)`. As: if it works, good, if it doesn't... bad luck 🤷‍♀️ 
Most of the time that should be ok 🤞 

The more reliable solutions are a bit more complicated, like using sidekiq batches and doing `on_success`... but IMO, for this, it is not worth it.